src/query_builder.test.ts

```diff
@@ -1,70 +1,172 @@
 import interpolateKustoQuery from './query_builder';
+import { TemplateSrv } from '@grafana/runtime';
+import { escapeSpecial } from 'datasource';
 
 describe('QueryBuilder', () => {
-  describe('when $__contains and multi template variable has custom All value', () => {
-    it('should generate a where..in clause', () => {
-      const query = interpolateKustoQuery('query=Tablename | where $__contains(col, all)');
+  const variableProps = (name: string, overrides?: any) => ({
+    name,
+    id: name,
+    multi: true,
+    includeAll: false,
+    type: 'custom',
+    ...overrides,
+  });
+  const variables = new Map<string, any>([
+    ['$__all', { ...variableProps('__all', { text: ['All'] }) }],
+    ['$test1', { ...variableProps('test1', { text: ['val1'] }) }],
```

Comment:
nit: rather than numbers, it would be better to have a description of the variable, for example `$test_one_var` `$test_two_vars`, `$test_quoted_var`...